libethereum/BlockChainSync.cpp

```diff
@@ -465,6 +465,15 @@ void BlockChainSync::onPeerBlockHeaders(std::shared_ptr<EthereumPeer> _peer, RLP
 			m_haveCommonHeader = true;
 			m_lastImportedBlock = (unsigned)info.number();
 			m_lastImportedBlockHash = info.hash();
+		
+			if (!m_headers.empty() && m_headers.begin()->first == m_lastImportedBlock + 1 && 
+				m_headers.begin()->second[0].parent != m_lastImportedBlockHash)
+			{
+				// Start of the header chain in m_headers doesn't match our known chain,
```

Comment:
How about also adding a log here to see how often this happens?

Reply:
Ok, I'll add it